src/components/Question.js

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+
+const Question = ({ question, answers, correct, id, checked, setSuccess }) => {
+  const [submitted, setSubmitted] = useState(null);
+
+  const responseElement = answers.map((response, index) => (
+    <div
+      key={index}
+      className={`border whitespace-nowrap border-primary-100  rounded-lg px-2 py-1 ${
+        submitted === index && !checked && 'bg-indigo-200'
+      } ${response === correct && checked && 'bg-green-300'} ${
+        submitted === index && correct !== response && checked && 'bg-red-100'
+      } `}
+    >
+      <button
+        disabled={checked}
+        onClick={() => {
+          setSubmitted(index);
+          setSuccess(prev => {
+            prev[id] = response === correct;
+            return prev;
+          });
+        }}
```

Comment:
I'd suggest putting this in a function outside this div and calling the function name in the `onClick` attribute, to improve the readability of your code and keep them clean.